### PR TITLE
OIDC auth filters to return raw ID Token 

### PR DESCRIPTION
**Pull request by anchalkhandelwal**
Request header parameter 'skipper-oidc-info' will also return raw ID token for the 3 filters OidcUserInfoName, OidcAnyClaimsName, OidcAllClaimsName. tokenContainer struct is extended for this.

Enhancement Request Issue #1448

### CODE REVIEW

**Comment by szuecs** [comment]
to fix DCO use `git commit -s`, you can use rebase to do this for all commits you did

**Comment by szuecs** [comment]
Please do a 
`git rebase -i master~5` and squash all commits into one, than you can do `git commit --amend --signoff ` for example to fix DCO

**Comment by anchalkhandelwal** [comment]
@szuecs Thanks for the help. That did solve the DCO issue and squash all commits into one. But now one of the test (check-race) is failing. Looking into it.

File: filters/auth/oidc.go [hunk 1]
```diff
@@ -512,7 +514,22 @@ func (f *tokenOidcFilter) callbackEndpoint(ctx filters.FilterContext) {
 
 			return
 		}
+		oidcIDToken, err = f.getidtoken(ctx, oauth2Token)
+		if err != nil {
+			if _, ok := err.(*requestError); !ok {
+				log.Errorf("Error while getting id token : %v.", err)
```
Comment (szuecs) [inline filters/auth/oidc.go:520]: nitpick: `token : %v.", err)` -> `token: %v", err)`

File: filters/auth/oidc.go [hunk 2]
```diff
@@ -512,7 +514,22 @@ func (f *tokenOidcFilter) callbackEndpoint(ctx filters.FilterContext) {
...
+			}
+
+			unauthorized(
+				ctx,
+				"",
+				invalidClaim,
+				r.Host,
+				fmt.Sprintf("Failed to get id token : %v.", err),
```
Comment (szuecs) [inline filters/auth/oidc.go:528]: nitpick: `token : %v.", err)` -> `token: %v", err)`

File: filters/auth/oidc.go [hunk 3]
```diff
@@ -526,6 +543,22 @@ func (f *tokenOidcFilter) callbackEndpoint(ctx filters.FilterContext) {
 			return
 		}
 	case checkOIDCAnyClaims, checkOIDCAllClaims:
+		oidcIDToken, err = f.getidtoken(ctx, oauth2Token)
+		if err != nil {
+			if _, ok := err.(*requestError); !ok {
+				log.Errorf("Error while getting id token : %v.", err)
```
Comment (szuecs) [inline filters/auth/oidc.go:549]: nitpick: `token : %v.", err)` -> `token: %v", err)`

File: filters/auth/oidc.go [hunk 4]
```diff
@@ -526,6 +543,22 @@ func (f *tokenOidcFilter) callbackEndpoint(ctx filters.FilterContext) {
...
+			}
+
+			unauthorized(
+				ctx,
+				"",
+				invalidClaim,
+				r.Host,
+				fmt.Sprintf("Failed to get id token : %v.", err),
```
Comment (szuecs) [inline filters/auth/oidc.go:557]: nitpick: `token : %v.", err)` -> `token: %v", err)`

**Comment by szuecs** [comment]
@anchalkhandelwal small nitpicks, but great work, thanks!

**Comment by szuecs** [comment]
@anchalkhandelwal the whitespace in front of the ` :` -> `:` please :)
Please don't care about the tests failures, these are known flaky tests and I will retrigger it and hopefully we can fix these flakytests some day.

**Comment by szuecs** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
Thanks @anchalkhandelwal !

**Pull request merged**